File: sample/src/main/java/com/github/lany192/box/sample/data/binder/ImageBinder.java
package com.github.lany192.box.sample.data.binder;

import androidx.annotation.NonNull;

import com.github.lany192.arch.items.ItemBinder;
import com.github.lany192.box.sample.databinding.ItemImageBinding;
import com.github.lany192.utils.ImageUtils;

public class ImageBinder extends ItemBinder<String, ItemImageBinding> {

    @Override
    public void bind(@NonNull ItemImageBinding binding, String url, int position) {
        ImageUtils.show(binding.image, url);
    }
}

File: sample/avatar/src/main/java/com/lany192/box/avatar/data/binder/ImageBinder.kt
package com.lany192.box.avatar.data.binder

import com.github.lany192.extension.load
import com.github.lany192.arch.items.ItemBinder
import com.lany192.box.avatar.databinding.ItemImageBinding

class ImageBinder : ItemBinder<String, ItemImageBinding>() {

    override fun convert(binding: ItemImageBinding, item: String, position: Int) {
        binding.image.load(item)
    }
}